superset/assets/visualizations/BigNumber.jsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { XYChart, AreaSeries, PointSeries, CrossHair, LinearGradient } from '@data-ui/xy-chart';
+
+import { brandColor } from '../javascripts/modules/colors';
+import { d3FormatPreset, d3TimeFormatPreset } from '../javascripts/modules/utils';
+import { getTextWidth } from '../javascripts/modules/visUtils';
+
+const fontFamily = '-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen,Ubuntu,Cantarell,Open Sans,Helvetica Neue,sans-serif';
+
+const CONTAINER_PADDING = 16;
+
+const CONTAINER_STYLES = {
+  fontFamily,
+  padding: CONTAINER_PADDING,
+  overflow: 'hidden',
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'center',
+};
+
+const BIG_NUMBER_STYLES = {
+  lineHeight: '1em',
+  paddingTop: '0.37em',
+  paddingBottom: '0.37em',
+  fontWeight: 600,
+};
+
+const SUBHEADER_STYLES = {
+  fontWeight: 200,
+  paddingTop: '0.5em',
+  paddingBottom: '0.5em',
+  lineHeight: '1em',
+};
+
+const CHART_MARGIN = {
+  top: 4,
+  right: 4,
+  bottom: 4,
+  left: 4,
+};
+
+const NUMBER_SIZE_SCALAR = 0.3;
+const CHART_SIZE_SCALAR = 0.2;
+const TEXT_SIZE_SCALAR = 0.1;
+
+// Returns the maximum font size (<= idealFontSize) that will fit within the available Width
+function getMaxFontSize({ text, availableWidth, idealFontSize, fontWeight = 'normal' }) {
```

Comment:
nit: Can we use arrow functions?
```
const getMaxFont = () => {...}
```

Reply:
I think it's generally preferable to define helper functions as normal functions as opposed to arrow functions because if there’s an error in `function myHelper` then the function name will appear in the stack trace whereas `const myHelper` will show up as an anonymous function.

I think arrow function style is preferable when you can take advantage of the syntax to make the function definition more succinct (like `const getDate = d => d.date`) but beyond that it seems like only upside to declaring a normal function.